crates/viewer/re_renderer/shader/world_grid.wgsl

```diff
@@ -0,0 +1,154 @@
+    let distance_to_grid_line = calc_distance_to_grid_line(in.scaled_world_plane_position);
+
+    // Figure out the how wide the lines are in this "draw space".
+    let plane_unit_pixel_derivative = fwidthFine(in.scaled_world_plane_position);
+    let line_anti_alias = plane_unit_pixel_derivative;
+    let width_in_pixels = 1.0;//config.thickness_ui * frame.pixels_from_point;
+    let width_in_grid_units = width_in_pixels * plane_unit_pixel_derivative;
+    var intensity_regular = linearstep2(width_in_grid_units + line_anti_alias, width_in_grid_units - line_anti_alias, distance_to_grid_line);
+
+    // Fade lines that get too close to each other.
+    // Once the number of pixels per line (== from one line to the next) is below a threshold fade them out.
+    //
+    // Note that `1/plane_unit_pixel_derivative` would give us more literal pixels per line,
+    // but we actually want to know how dense the lines get here so we use `1/width_in_grid_units` instead,
+    // such that a line density of 1.0 means roughly "100% lines" and 10.0 means "Every 10 pixels there is a lines".
+    // Empirically (== making the fade a hard cut and taking screenshot), this works out pretty accurately!
+    //
+    // Tried smoothstep here, but didn't feel right even with lots of range tweaking.
+    let line_density = 1.0 / max(width_in_grid_units.x, width_in_grid_units.y);
+    let grid_closeness_fade = linearstep(1.0, 10.0, line_density);
+    intensity_regular *= grid_closeness_fade;
+
+    // Every tenth line is a more intense, we call those "cardinal" lines.
+    // Experimented previously with more levels of cardinal lines, but it gets too busy:
+    // It seems that if we want to go down this path, we should ensure that there's only two levels of lines on screen at a time.
+    const CARDINAL_LINE_FACTOR: f32 = 10.0;
+    let distance_to_grid_line_cardinal = calc_distance_to_grid_line(in.scaled_world_plane_position * (1.0 / CARDINAL_LINE_FACTOR));
+    var cardinal_line_intensity = linearstep2(width_in_grid_units + line_anti_alias, width_in_grid_units - line_anti_alias,
+                                              distance_to_grid_line_cardinal * CARDINAL_LINE_FACTOR);
+    let cardinal_grid_closeness_fade = linearstep(2.0, 10.0, line_density * CARDINAL_LINE_FACTOR); // Fade cardinal lines a little bit earlier (because it looks nicer)
+    cardinal_line_intensity *= cardinal_grid_closeness_fade;
+
+    // Combine all lines.
+    //
+    // Lerp for cardinal & regular.
+    // This way we don't break anti-aliasing (as addition would!), mute the regular lines, and make cardinals weaker when there's no regular to support them.
+    let cardinal_and_regular = mix(intensity_regular, cardinal_line_intensity, 0.4);
+    // X and Y are combined like akin to premultiplied alpha operations.
+    let intensity_combined = saturate(cardinal_and_regular.x * (1.0 - cardinal_and_regular.y) + cardinal_and_regular.y);
```

Comment:
tbh not so sure about this anymore. Sounded like a compelling idea in Ben's article, but looking at the close-ups I actually don't like the change in color when grid lines meet.
So maybe more like

```suggestion
    let intensity_combined = max(cardinal_and_regular.x, cardinal_and_regular.y);
```

It's very subtle though either way.

Reply:
> `let intensity_combined = saturate(cardinal_and_regular.x * (1.0 - cardinal_and_regular.y) + cardinal_and_regular.y);`

<img width="673" alt="Screenshot 2024-11-27 at 21 37 30" src="https://github.com/user-attachments/assets/076172d6-a866-4192-8d38-5b0a999af7c4">

----

> `let intensity_combined = max(cardinal_and_regular.x, cardinal_and_regular.y);`

<img width="667" alt="Screenshot 2024-11-27 at 21 36 58" src="https://github.com/user-attachments/assets/fd5f35ef-1588-40c3-9f0d-37da3de78f21">


----

I think the first is better at this particular angle (less [Moiré](https://xkcd.com/1814/)). Hard to test without camera control

Reply:
oh wow yeah so much better, agreed
more visual refinements on the way in the follow-up pr